### PR TITLE
Bk-LSTM

### PR DESCRIPTION
**Pull request by evanyeohboonkhai**
Please check all the file name start with LSTM

### CODE REVIEW

**Review by KianYang-Lee — CHANGES_REQUESTED** [review]
Error in 04 - Univariate Single Step LSTM.ipynb. Please correct.

<code>NameError                                 Traceback (most recent call last)
<ipython-input-29-303829621deb> in <module>
      4 
      5 # Build Scaler and scale the train and test data
----> 6 scaler = MinMaxScaler(feature_range=(-1, 1))
      7 train_data_normalized_shampoo = scaler.fit_transform(train_data_shampoo)
      8 test_data_normalized_shampoo = scaler.fit_transform(test_data_shampoo)

NameError: name 'MinMaxScaler' is not defined</code>

**Comment by evanyeohboonkhai** [comment]
Error in 04 - Univariate Single Step LSTM.ipynb.  solved 
This is a duplicated file, hence I remove it. Thanks @KianYang-Lee

File: notebook/Solution/04 - LSTM -Multivariate Input -Univariate Output-MultiStep.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM -Multivariate Input -Univariate Output-MultiStep.ipynb]: 2nd colon kinda redundant if you are not specifying any value? Maybe remove it?

File: notebook/Solution/04 - LSTM -Multivariate Input -Univariate Output-MultiStep.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM -Multivariate Input -Univariate Output-MultiStep.ipynb]: Again, believe you can come up with a better explanation. Thanks. Now it is not really intuitive.

**Review by KianYang-Lee — CHANGES_REQUESTED** [review]
Looked at the flow and logic of the codes, did not check the Markdown so will be suggesting based on flow and logic soundness, which were feedback in each file. General comments are:

1. A lot of repetitive parts across notebooks, it would be helpful to put them in a separate module and just call them as methods.
2. Some showcase of basic data analytics would be helpful to prime the viewers on how to perform effective data analysis, and also to prep them for the exam since we are going to check with them on that part.
3. Model performance wise looks like better compared to CNN.

Overall, a job well done. Hope that you can make some modifications based on the input and we will be ready to merge after that basing on the code logic.

**Review by scot-certifai — APPROVED** [review]

**Comment by KianYang-Lee** [comment]
image
Awaiting modifications from branch owner before merging. Avoid merging first to all until confirmation from @evanyeohboonkhai  thanks

**Review by KianYang-Lee — APPROVED** [review]

**Comment by scot-certifai** [comment]
Can this branch be merged already @evanyeohboonkhai ?

**Comment by evanyeohboonkhai** [comment]
> Can this branch be merged already @evanyeohboonkhai ?

not yet, currently, update with the instruction

**Comment by evanyeohboonkhai** [comment]
please have a review in the following LSTM file 
1. 04-LSTM-Univariate-Single-Step.ipynb
2. 04-LSTM-Univariate-Multi-Step.ipynb
3. 04-LSTM-Multivariate-Input-Univariate-Output-Single-Step.ipynb
4. 04-LSTM-Multivariate-Input-Univariate-Output-Multi-Step.ipynb

Add in data_module.py , deep_learning_module.py to modularize the function require 

Remove Multivariate input , multivariate output.ipynb due to need more time research , will include in future

File: notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Multi-Step.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Multi-Step.ipynb:2963]: I think this a mistake of strikethrough the other notebooks?

File: notebook/Solution/04 - LSTM-Univariate-Multi-Step.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM-Univariate-Multi-Step.ipynb:1832]: TypeError, please resolve
image

File: notebook/Solution/04 - LSTM-Univariate-Single-Step.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM-Univariate-Single-Step.ipynb]: People either refer to feature as "feature" or "x", and label as "label" or "y"

File: notebook/Solution/04 - LSTM-Univariate-Single-Step.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM-Univariate-Single-Step.ipynb]: Pelase remove empty cell

File: .gitignore
Comment (KianYang-Lee) [inline .gitignore:7]: Can remove redundant lines

File: notebook/Solution/data_module.py
Comment (KianYang-Lee) [inline notebook/Solution/data_module.py]: Remove In[3] and other redundant lines

File: notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Single-Step.ipynb
Comment (KianYang-Lee) [inline notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Single-Step.ipynb:64]: suggest to include torchsummaryX to environment

**Review by KianYang-Lee — APPROVED** [review]
Great job, made some suggestions to further improve the slide

File: notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Single-Step.ipynb
Comment (evanyeohboonkhai) [reply notebook/Solution/04 - LSTM-Multivariate-Input-Univariate-Output-Single-Step.ipynb]: image
cannot found torchsummaryX in conda environment, hence use pip install

**Review by KianYang-Lee — APPROVED** [review]
Already reviewed and approved

**Comment by evanyeohboonkhai** [comment]
@KianYang-Lee  Please merge if it's ok for you

**Review by scot-certifai — APPROVED** [review]

**Pull request merged**
